src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,45 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_LENGTH = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_LENGTH];
+        values = (V[]) new Object[MAX_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (get(key) != null) {
+                values[i] = value;
+                return;
+            }
```

Comment:
let's corret your if statement a little, otherwise it makes no sense
```suggestion
        for (int i = 0; i < size; i++) {
            if (key == keys[i] || key != null && key.equals(keys[i])) {
                values[i] = value;
                return;
            }
```